languages/rust/exercises/concept/string-vs-str/.docs/introduction.md

```diff
@@ -0,0 +1,56 @@
+`String` is a potentially-mutable utf8-encoded representation of a sequence of Unicode codepoints. (In Java or C#, this type is called `StringBuilder`.)
+
+`&str` is a read only view of a well-formed utf8 sequence and because it's a reference it is `Copy` and can be shared.
+
+`String` and `&str` in rust can not be indexed into like you might in other languages.
+For example this will not work:
+
+```rust,invalid
```

Comment:
What does the `invalid` do? Never seen that used before, interesting!

Reply:
The only place I'm really familiar with seeing multiple tags on a code fence like that is when writing Rust documentation; tagging a snippet as "invalid" will render it with a red box surrounding it and a popup noting that this is invalid. It also won't attempt to compile it as a doctest. 

That said, I did check; the GFM renderer appears to properly render the sample as Rust, which is sufficient.

Reply:
Thanks for the explanation!